cpp/node.h

```diff
@@ -0,0 +1,78 @@
+#ifndef NODE_H_
+#define NODE_H_
+
+class Node {
```

Comment:
Add Doxygen style comments for the class, what's the class supposed to do. Look at cb-geo/mpm or cb-geo/lem for examples.